src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,80 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = todo;
+
+  const [user, setUser] = useState<User | null>(null);
+  const dispatch = useAppDispatch();
+
+  const handleCloseClick = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
+
+  useEffect(() => {
+    getUser(userId)
```

Comment:
There is no error handling for the promise returned by the getUser function. If the promise rejects, the error will be unhandled. Add a .catch() block to handle any potential errors